horace_core/sqw/PixelData/@PixelData/private/binary_op_double_.m

```diff
@@ -0,0 +1,110 @@
+function obj = binary_op_double_(obj, double_array, binary_op, flip, npix)
+%% BINARY_OP_DOUBLE_ perform a binary operation between this PixelData object
+% and an array
+%
+validate_input_array(obj, double_array, npix);
+
+if isempty(npix)
+    base_page_size = obj.max_page_size_;
+    while true
+
+        pix_sigvar = sigvar(obj.signal, obj.variance);
+
+        start_idx = (obj.page_number_ - 1)*base_page_size + 1;
+        end_idx = min(start_idx + base_page_size - 1, obj.num_pixels);
+
+        double_sigvar = sigvar(double_array(start_idx:end_idx), []);
+        [obj.signal, obj.variance] = ...
+                sigvar_binary_op_(pix_sigvar, double_sigvar, binary_op, flip);
+
+        if obj.has_more()
+            obj = obj.advance();
+        else
+            break;
+        end
+
+    end
+else
+
+    npix_cum_sum = cumsum(npix(:));
+    if npix_cum_sum(end) ~= obj.num_pixels
+        error('PIXELDATA:binary_op_double_', ...
```

Comment:
is this the error the `different npix numbers` error above is pre-catching?

Reply:
The `different npix numbers` error is for sqw-sqw operations, whereas this is for sqw-double.

Reply:
Is this referring to the `validate_input_array` function? That only deals with the case where `npix` is empty. This catches the case where `npix` is not empty but `sum(npix) != pix.num_pixels`. Perhaps this check could go into the `validate_` function and return `npix_cum_sum`.